### PR TITLE
Add aria-label to icon only menu button stories

### LINKED ISSUE #1511: [menu-button] Missing `aria-label` attributes on icon only stories
### Provide a general summary of the issue here

See title

### 🤔 Expected Behavior?

-

### 😯 Current Behavior

-

### 💁 Possible Solution

- Add a `label` property to `<sl-menu-button>` and apply that as `aria-label` on the nested button
- Log a warning when the button doesn't have a label or aria-label property

### 🖥️ Steps to Reproduce

-

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### If other, please specify.

_No response_

### What operating system are you using?

macOS

### 👤 Your name

Jeroen Zwartepoorte

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning

### PR DESCRIPTION
**Pull request by jpzwarte**
Fixes #1511

### CODE REVIEW

File: packages/components/menu/src/menu-button.ts [hunk 1]
```diff
@@ -51,6 +51,9 @@ export class MenuButton extends ScopedElementsMixin(LitElement) {
   /** The fill of the button. */
   @property() fill: ButtonFill = 'outline';
 
+  /** If the menu button is icon only, use this to set the `aria-label` on the button. */
+  @property() label?: string;
```
Comment (anna-lach) [inline packages/components/menu/src/menu-button.ts:55]: Are we sure that this is really necessary? Adding `aria-label` on the `menu-button` is enough for the Voice Over, like in this example:
https://proud-water-0d02ee203-1541.westeurope.2.azurestaticapps.net/categories/components/menu/usage/

Maybe this PR should only cover adding `aria-label` in the stories  of menu button - icon only?

Comment (anna-lach) [reply packages/components/menu/src/menu-button.ts:55]: If we really want to keep it this way, we also need to change the documentation about `aria-label` https://proud-water-0d02ee203-1541.westeurope.2.azurestaticapps.net/categories/components/menu/accessibility/ in the menu button.

Comment (Diaan) [reply packages/components/menu/src/menu-button.ts:55]: Agreed; if we force the use of `aria-label` we sort of remove the option of using `aria-labeled-by` when that is more appropriate.
If a dev doesn't think about using `aria-label` they probably also won't think to use `label`

Comment (jpzwarte) [reply packages/components/menu/src/menu-button.ts]: Changed the PR to now only add an `aria-label` to the icon only stories

**Pull request merged**